sources/chargen.js

```diff
@@ -665,27 +667,33 @@ $(document).ready(function() {
 
         layers = []
         const previewToDraw = {};
-        previewToDraw.link = $(this).data(`layer_1_${getBodyTypeName()}`);
-        previewToDraw.zPos = $(this).data(`layer_1_zpos`);
-        layers.push(previewToDraw);
-        
-        for(jdx =2; jdx < 10; jdx++){
-          if($(this).data(`layer_${jdx}_${getBodyTypeName()}`)){
-            const previewToDraw = {};
-            previewToDraw.link = $(this).data(`layer_${jdx}_${getBodyTypeName()}`);
-            previewToDraw.zPos = $(this).data(`layer_${jdx}_zpos`);
-
-            layers.push(previewToDraw);
-          } else {
-            break;
-          }
-        }    
+        const animation =  $(this).data(`layer_1_custom_animation`);
+
+        if($(this).data(`layer_1_${getBodyTypeName()}`) === undefined){
+          previewToDraw.link = $(this).data(`layer_1_${getBodyTypeName()}`);
+          previewToDraw.zPos = $(this).data(`layer_1_zpos`);
+          layers.push(previewToDraw);
+        }
+        else{
```

Comment:
could you move this else on the closing bracket of the above if?

Reply:
I think I fixed it